cosmos/runtime/app.go

```diff
@@ -309,8 +310,9 @@ func NewPolarisApp( //nolint: funlen // from sdk.
 	// ===============================================================
 
 	// Polaris Configuration
-	cfg := provider.DefaultConfig()
-	cfg.NodeConfig.DataDir = DefaultNodeHome + "/data"
+	homePath := GetHomePath(appOpts)
```

Comment:
Like pass the base string down to the provider and have the provider do all the parsing and stuff